include/PathUtil.h

```diff
@@ -0,0 +1,30 @@
+#ifndef PATHUTIL_H
+#define PATHUTIL_H
+
+#include "lmms_export.h"
+
+#include <QDir>
+
+namespace PathUtil
+{
+	enum Base { AbsoluteBase, ProjectDirBase, FactorySampleBase, UserSampleBase, UserVSTBase, PresetBase,
+		UserLADSPABase, DefaultLADSPABase, UserSoundfontBase, DefaultSoundfontBase, UserGIGBase, DefaultGIGBase };
+
+	QString LMMS_EXPORT baseLocation(Base base);
+	QDir LMMS_EXPORT baseQDir (Base base);
+	QString LMMS_EXPORT basePrefix(Base base);
+	Base LMMS_EXPORT baseLookup(QString path);
```

Comment:
Pass by value involves unnecessary data copying. Use pass by references if you can.
For example:
```suggestion
	Base LMMS_EXPORT baseLookup(const QString & path);
```
If you do so, please don't forget to change the signature in the definition as well as the one in the declaration!